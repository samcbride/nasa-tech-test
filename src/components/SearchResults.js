import React from "react";
import "../styles/SearchResults.css";
import PropTypes from "prop-types";

const SearchResults = ({ searchResults }) => {
  if (!searchResults) {
    return <p>No results</p>;
  } else {
    return (
      <>
        {searchResults.map((image) => (
          <img
            key={image}
            className="card-image"
            src={image}
            alt="spaceImage"
          />
        ))}
      </>
    );
  }
};

export default SearchResults;

SearchResults.propTypes = {
  searchResults: PropTypes.array.isRequired,
};
