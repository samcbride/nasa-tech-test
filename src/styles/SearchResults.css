.card-image {
  display: flex;
  flex-direction: row;
  flex-wrap: wrap;
  width: 17em;
  margin: 1em;
  border-radius: 2%;
}

/* img { 
  height: 12em;
  width: 17em;
  margin: 1em;
  border-radius: 2%;
} */
