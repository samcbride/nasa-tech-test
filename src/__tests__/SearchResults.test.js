import React from "react";
import { render } from "@testing-library/react";
import SearchResults from "../components/SearchResults";

describe("SearchResults", () => {
  const stub = ["image1.jpeg", "image2.jpeg", "image3.jpeg"];
  const { asFragment } = render(<SearchResults searchResults={stub} />);

  it("renders correctly", () => {
    expect(asFragment()).toMatchSnapshot();
  });

  it("renders alt text correctly", () => {
    const { getAllByAltText } = render(<SearchResults searchResults={stub} />);
    expect(getAllByAltText("spaceImage")[0]).toHaveClass("card-image");
  });
});
