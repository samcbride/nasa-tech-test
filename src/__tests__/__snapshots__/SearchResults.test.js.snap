// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchResults renders correctly 1`] = `
<DocumentFragment>
  <img
    alt="spaceImage"
    class="card-image"
    src="image1.jpeg"
  />
  <img
    alt="spaceImage"
    class="card-image"
    src="image2.jpeg"
  />
  <img
    alt="spaceImage"
    class="card-image"
    src="image3.jpeg"
  />
</DocumentFragment>
`;
